exporter/elasticsearchexporter/config.go

```diff
@@ -319,4 +322,7 @@ func logConfigDeprecationWarnings(cfg *Config, logger *zap.Logger) {
 	if !cfg.Mapping.Dedup {
 		logger.Warn("dedup has been deprecated, and will always be enabled in future")
 	}
+	if !cfg.Mapping.Dedot && cfg.MappingMode() == MappingECS {
+		logger.Warn("dedot has been deprecated, and will always be enabled in ECS mode in future")
+	}
```

Comment:
Does it make sense to also warn in the opposite case?

```suggestion
	if cfg.Mapping.Dedot && cfg.MappingMode() != MappingECS {
		logger.Warn("dedot has been deprecated, and will always be disabled in the future, unless in ECS mode")
	}
	if !cfg.Mapping.Dedot && cfg.MappingMode() == MappingECS {
		logger.Warn("dedot has been deprecated, and will always be enabled in ECS mode in future")
	}
```

Reply:
Maybe... the reason I didn't do that is because this is the default. So we would be warning by default. OTOH maybe that's desirable, to get folks prepared.

Reply:
Done in e28ba42c076b392cba0bd4d079119276ec80678f

Reply:
I'd prefer noisy warnings over surprises. The suggestion makes sense to me. It'll be even better if we could consolidate on the warning messages and log the same warning in both scenarios,.